lib/chef/resource/file.rb

```diff
@@ -115,6 +117,18 @@ def manage_symlink_source(arg=nil)
           :kind_of => [ TrueClass, FalseClass ]
         )
       end
+
+      def verify(command=nil, opts={}, &block)
+        if ! (command.nil? || [String, Symbol].include?(command.class))
+          raise ArgumentError, "verify requires either a string, symbol, or a block"
```

Comment:
A user could call the no-arg, no-block `verify` and it would make it past this argument validation - seems to me that shouldn't be possible.